mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -418,21 +418,21 @@ trait Params extends Identifiable with Serializable {
   /**
    * Sets a parameter in the embedded param map.
    */
-  protected final def set[T](param: Param[T], value: T): this.type = {
```

Comment:
The one that I could think of is if someone was trying to interact with the object based on which params it had this might be easier, but thats probably not enough of a reason. I can make this protected and just leave clear public if that sounds good.